### PR TITLE
return T::Boolean for Module.=== with an untyped arg

### LINKED ISSUE #4173: Module#=== drops type information when argument is untyped
#### Input

[→ View on sorbet.run](https://sorbet.run/#%23%20typed%3A%20true%0Aextend%20T%3A%3ASig%0A%0Ax%20%3D%20T.let(nil%2C%20T.nilable(Integer))%0AT.reveal_type(NilClass.%3D%3D%3D(x))%0A%0AT.reveal_type(NilClass.%3D%3D%3D(T.unsafe(nil))))

```ruby
# typed: true
extend T::Sig

x = T.let(nil, T.nilable(Integer))
T.reveal_type(NilClass.===(x))

T.reveal_type(NilClass.===(T.unsafe(nil)))
```

#### Observed output

```
editor.rb:5: Revealed type: T::Boolean https://srb.help/7014
     5 |T.reveal_type(NilClass.===(x))
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  Got T::Boolean originating from:
    editor.rb:5:
     5 |T.reveal_type(NilClass.===(x))
                      ^^^^^^^^^^^^^^^

editor.rb:7: Revealed type: T.untyped https://srb.help/7014
     7 |T.reveal_type(NilClass.===(T.unsafe(nil)))
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  Got T.untyped originating from:
    editor.rb:7:
     7 |T.reveal_type(NilClass.===(T.unsafe(nil)))
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^
Errors: 2
```

#### Expected behavior

There's no reason why the argument being `T.untyped` should affect the return type. (It would be another story if the receiver of the `===` was untyped.)

In this case, the revealed types should both be `T::Boolean`.

- - -

### PR DESCRIPTION
**Pull request by froydnj**
### Motivation

Fixes #4173.

### Test plan

See included automated tests.

### CODE REVIEW

**Review by elliottt — APPROVED** [review]
This makes sense to me!

**Review by elliottt — APPROVED** [review]
This makes sense to me!

**Comment by froydnj** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_JtwDvXsvOPSf6y
→ https://go/builds/bui_JtwD3d2q1EmuJk

File: core/types/calls.cc [hunk 1]
```diff
@@ -3057,7 +3057,7 @@ class Module_tripleEq : public IntrinsicMethod {
         }
         auto rhs = args.args[0]->type;
         if (rhs.isUntyped()) {
-            res.returnType = rhs;
+            res.returnType = Types::Boolean();
```
Comment (jez) [inline core/types/calls.cc:3060]: lol, what 😅 thanks for fixing this 🙈

**Pull request merged**